sdk/common/src/main/java/io/opentelemetry/sdk/common/export/ConfigBuilder.java

```diff
@@ -158,4 +159,91 @@ protected static Double getDoubleProperty(String name, Map<String, String> map)
       return null;
     }
   }
+
+  /**
+   * Get a duration property from the map, {@code null} if it cannot be found or it has a wrong
+   * type.
+   *
+   * <p>Durations can be of the form "{number}{unit}", where unit is one of:
+   *
+   * <ul>
+   *   <li>ms
+   *   <li>s
+   *   <li>m
+   *   <li>h
+   *   <li>d
+   * </ul>
+   *
+   * @param name The property name
+   * @param map The map where to look for the property
+   * @return the {@link Long} value of the property in milliseconds, {@code null} in case of error
+   *     or if the property cannot be found.
+   */
+  @Nullable
+  protected static Long getDurationMillisProperty(String name, Map<String, String> map) {
+    try {
+      String value = map.get(name);
+      String unitString = getDurationUnitString(value);
+      TimeUnit units = getDurationUnit(unitString);
+      String numberString = value.substring(0, value.length() - unitString.length()).trim();
+      long rawNumber = Long.parseLong(numberString);
+      return TimeUnit.MILLISECONDS.convert(rawNumber, units);
+    } catch (NumberFormatException | NullPointerException ex) {
+      return null;
+    }
+  }
+
+  /** Returns the TimeUnit associated with a unit string. Defaults to milliseconds. */
+  private static TimeUnit getDurationUnit(String unitString) {
+    // If statement in likely-to-be-used order.
+    if ("s".equals(unitString)) {
+      return TimeUnit.SECONDS;
+    } else if ("ms".equals(unitString)) {
+      return TimeUnit.MILLISECONDS;
+    } else if ("m".equals(unitString)) {
+      return TimeUnit.MINUTES;
+    } else if ("h".equals(unitString)) {
+      return TimeUnit.HOURS;
+    } else if ("d".equals(unitString)) {
+      return TimeUnit.DAYS;
+    } else if ("".equals(unitString)) {
+      return TimeUnit.MILLISECONDS;
+    }
+    throw new NumberFormatException("Invalid duration string, found: " + unitString);
+  }
+
+  /**
+   * Fragments the 'units' portion of a config value from the 'value' portion.
+   *
+   * <p>E.g. "1ms" would return the string "ms".
+   */
+  private static String getDurationUnitString(String rawValue) {
```

Comment:
Not a big difference, but I think I would search for the first non-number to avoid the additional `isDurationUnitToken` method.

Reply:
I moved to that. There's a lot of "oddities" w/ i18n and codepoints here, depending on what people paste in.  LIKELY we should throw a ConfigException on most attempts to hack this value, but my brain's still not quite up to the task of remembering the problematic codepoints + encodings to try out in tests.